bin/yaml/java.yaml

```diff
@@ -51,3 +51,8 @@ compilers:
       - name: 22.0.0
         url: https://download.java.net/java/GA/jdk22/830ec9fcccef480bb3e73fb7ecafe059/36/GPL/openjdk-22_linux-x64_bin.tar.gz
         untar_dir: jdk-22
+      - name: 22.0.2
+        url: https://download.java.net/java/GA/jdk22.0.2/c9ecb94cd31b495da20a27d4581645e8/9/GPL/openjdk-22.0.2_linux-x64_bin.tar.gz
+      - name: 23.0.1
+        url: https://download.java.net/java/GA/jdk23.0.1/c28985cbf10d4e648e4004050f8781aa/11/GPL/openjdk-23.0.1_linux-x64_bin.tar.gz
+        untar_dir: jdk-23
```

Comment:
After a quick inspection of the archives, I believe that the untar_dir parameters should be changed.

- jdk 22.0.2 has a directory named jdk-22.0.2 as seen below
![image](https://github.com/user-attachments/assets/bf027454-ae56-4ee9-99cf-3f6ccf516dbe)

- jdk 23 has a directory named jdk-23.0.1 as seen below
![image](https://github.com/user-attachments/assets/a3f764d3-a084-43e3-85ad-ea446a1f2626)

Reply:
Thanks for taking a look at it and bringing my attention to this issue.
If I'm not mistaken, the jdk 22.0.2 is correct the way it is, doesn't need `untar_dir` added, since `dir` is specified like this, and the directory is corresponding to this:
https://github.com/compiler-explorer/infra/blob/3b37599e9bba2cc7d2d300b974b81f069ae07a51/bin/yaml/java.yaml#L5

Following this logic, removed the `untar_dir` from jdk-23.0.1, where the name of the directory is also corresponding to this. Before this change, the `ce_installl install java` indeed failed for jdk 23.0.1, but now it's successful on my local machine.

Please LMK if I missed something and it's still not OK.

Reply:
You're correct, just an oversight on my end